core/server/api_container/server/startosis_engine/kurtosis_instruction/kurtosis_instruction.go

```diff
@@ -34,5 +25,10 @@ type KurtosisInstruction interface {
 	ValidateAndUpdateEnvironment(environment *startosis_validator.ValidatorEnvironment) error
 
 	// TryResolveWith assesses whether the instruction can be resolved with the one passed as an argument.
-	TryResolveWith(other KurtosisInstruction, enclaveComponents *enclave_structure.EnclaveComponents) enclave_structure.InstructionResolutionStatus
+	TryResolveWith(other *enclave_plan_persistence.EnclavePlanInstruction, enclaveComponents *enclave_structure.EnclaveComponents) enclave_structure.InstructionResolutionStatus
+
+	// GetPersistableAttributes returns an EnclavePlanInstruction object which contains all the persistable attributes
+	// for this instruction. Persistable attributes are what will be written to the enclave database so that even
+	// if the APIC is restarted, idempotent runs will continue to work.
+	GetPersistableAttributes() *enclave_plan_persistence.EnclavePlanInstructionBuilder
```

Comment:
I think there is a mix between the function name, the comment, and the returned object type. The method name says that will return the Persitable attributes, but the returned method type is a builder and the comment says it will return an `EnclavePlanInstruction` I understand how it works but I think that it could be a bit confusing for devs